streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobScheduler.scala

```diff
@@ -40,7 +40,9 @@ private[scheduler] case class ErrorReported(msg: String, e: Throwable) extends J
 private[streaming]
 class JobScheduler(val ssc: StreamingContext) extends Logging {
 
-  private val jobSets = new ConcurrentHashMap[Time, JobSet]
+  // Use of ConcurrentHashMap.keySet later causes an odd runtime problem due to Java 7/8 diff
+  // https://gist.github.com/AlainODea/1375759b8720a3f9f094
+  private val jobSets: java.util.Map[Time, JobSet] = new ConcurrentHashMap[Time, JobSet]
```

Comment:
Not strictly a compiler warning, but an error I ran into while testing. Strange, subtle cross-compile issue with an easy fix; see https://gist.github.com/AlainODea/1375759b8720a3f9f094